src/mlpack/methods/ann/layer/sampling_impl.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file sampling_impl.hpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Sampling class which samples from parameters for a given
+ * distribution.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SAMPLING_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SAMPLING_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "sampling.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+Sampling<InputDataType, OutputDataType>::Sampling()
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+Sampling<InputDataType, OutputDataType>::Sampling(
```

Comment:
Is there any specific case in which this constructor will be used? I think the constructor which accepts only the size should suffice.